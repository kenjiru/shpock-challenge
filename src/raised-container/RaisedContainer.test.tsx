import * as React from "react";
import { CSSProperties, HTMLProps } from "react";
import { MuiThemeProvider } from "material-ui/styles";
import { mount, ReactWrapper, shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";

import ThemedRaisedContainer, { IRaisedContainerProps, RaisedContainer } from "./RaisedContainer";

describe("RaisedContainer", () => {
    it("should render correctly", () => {
        const output = shallow(
            <RaisedContainer
                className="custom-class"
                checked={false}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("should have the correct style if is checked", () => {
        const output: ReactWrapper = mount(
            <MuiThemeProvider>
                <ThemedRaisedContainer checked={true}/>
            </MuiThemeProvider>
        );

        const raisedContainerProps: IRaisedContainerProps = output.find(RaisedContainer).props();
        const divProps: HTMLProps<HTMLElement> = output.find(RaisedContainer).childAt(0).props();

        expect(divProps.style.backgroundColor).toBe(raisedContainerProps.muiTheme.palette.primary1Color);
    });

    it("should have the correct style if is NOT checked", () => {
        const output: ReactWrapper = mount(
            <RaisedContainer checked={false}/>
        );

        let style: CSSProperties = output.childAt(0).props().style;
        expect(style.backgroundColor).toBe(RaisedContainer.NORMAL_BACKGROUND);
    });
});
