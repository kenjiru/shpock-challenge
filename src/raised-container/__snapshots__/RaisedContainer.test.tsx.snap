// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaisedContainer should render correctly 1`] = `
<div
  className="raised-container custom-class"
  style={
    Object {
      "backgroundColor": "#e0e0e0",
    }
  }
/>
`;
